server/src/util/data-scheme.util.ts

```diff
@@ -1,24 +1,102 @@
-import { Injectable, InternalServerErrorException } from "@nestjs/common";
+import { Injectable, NotFoundException } from "@nestjs/common";
 import { Neo4jService } from "nest-neo4j/dist";
-import { DataSchemeService } from "../data-scheme/data-scheme.service";
 import Node from "../nodes/node.model";
-import { Attribute } from "../data-scheme/models/attributes";
-import MandatoryAttributeMissingException from "./exceptions/MandatoryAttributeMissing.exception";
+import { Attribute } from "../data-scheme/models/attributes.model";
 import Relation from "../relations/relation.model";
-import { RelationType } from "../data-scheme/models/relationType";
-import { LabelScheme } from "../data-scheme/models/labelScheme";
+import { RelationType } from "../data-scheme/models/relation-type.model";
+import { LabelScheme } from "../data-scheme/models/label-scheme.model";
+import { Connection } from "../data-scheme/models/connection.model";
+import { DatabaseUtil } from "./database.util";
+import { Datatype } from "../data-scheme/models/data-type.model";
 
 @Injectable()
 export class DataSchemeUtil {
-    constructor(private readonly neo4jService: Neo4jService, private readonly dataSchemeService: DataSchemeService) {}
+    /**
+     * The name of the tool database
+     */
+    private readonly database = process.env.DB_TOOL;
+
+    constructor(private readonly neo4jService: Neo4jService, private readonly databaseUtil: DatabaseUtil) {}
+
+    /**
+     * Parses the record to labels
+     *
+     * @param record Single record response from db
+     */
+    public parseLabelScheme(record: Record<any, any>): LabelScheme {
+        const l = {
+            ...record.get("dataScheme"),
+        };
+        l.attributes = JSON.parse(l.attributes, Attribute.reviver);
+        return Object.assign(new LabelScheme(), l);
+    }
+
+    /**
+     * Parses the record to relationTypes
+     *
+     * @param record Single record response from db
+     */
+    public parseRelationType(record: Record<any, any>): RelationType {
+        const l = {
+            ...record.get("dataScheme"),
+        };
+
+        l.attributes = JSON.parse(l.attributes, Attribute.reviver);
+        l.connections = JSON.parse(l.connections).map((conn) => Object.assign(new Connection(), conn));
+
+        return Object.assign(new RelationType(), l);
+    }
+
+    /**
+     * Parse the db response into a Node
+     *
+     * @param record Single record response from db
+     * @param labelScheme The label scheme of the node
+     */
+    public async parseNode(record, labelScheme?: LabelScheme): Promise<Node> {
+        const attributes = record.get("node");
+
+        const node = {
+            nodeId: record.get("node").nodeId,
+            name: record.get("node").name,
+            label: record.get("node").label,
+            attributes: attributes,
+        } as Node;
+
+        delete node["attributes"]["nodeId"];
```

Comment:
NodeId shouldn't be deleted from attributes, if it is specified in the data scheme, it should be transmitted.